background-processor/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+use lightning::chain;
+use lightning::chain::chaininterface::{BroadcasterInterface, FeeEstimator};
+use lightning::chain::keysinterface::{ChannelKeys, KeysInterface};
+use lightning::ln::channelmanager;
+use lightning::ln::channelmanager::ChannelManager;
+use lightning::{log_internal, log_error, log_trace};
+use lightning::util::logger::Logger;
+use lightning::util::ser::Writeable;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::thread;
+use std::time::{Duration, Instant};
+
+/// BackgroundProcesser runs a background thread that monitors whether the ChannelManager it's given
+/// on start has new updates to persist. If it does, then it writes said ChannelManager to
+/// disk/backups.
+/// If ChannelManager persistence fails and the persisted manager becomes out-of-date, then there is
+/// a risk of channels force-closing on startup when the manager realizes it's outdated. However, as
+/// long as `ChannelMonitor` backups are sound, no funds besides those used for unilateral chain
+/// closure fees are at risk.
+pub struct BackgroundProcessor {
+	stop_thread: Arc<AtomicBool>,
+}
+
+// Re-export these modules for the logging macros.
+mod util {
+	pub(crate) mod logger {
+		pub(crate) use lightning::util::logger::Record;
+	}
+}
+
+#[cfg(not(test))]
+const CHAN_FRESHNESS_TIMER: u64 = 60;
+#[cfg(test)]
+const CHAN_FRESHNESS_TIMER: u64 = 1;
+
+pub type PersistErrorCallback = fn(std::io::Error);
+
+impl BackgroundProcessor {
+	/// Start the background thread that checks whether new updates are available for the ChannelManager,
+	/// and writes it to disk if so.
+	/// We also call ChannelManager::timer_chan_freshness_every_min(), since we're already eating the
+	/// cost of running a thread, we may as well pack as many operations into it as make sense.
```

Comment:
This sentence reads a bit more like a design decision than user-facing documentation. Consider leaving it out since the more general nature of the struct name should make it clear it is responsible for other tasks. The call to `ChannelManager::timer_chan_freshness_every_min` can be mentioned in the struct documentation when enumerating its responsibilities.